### ISSUE TITLE
misspell can't be excluded if line is too long

### ISSUE THREAD

**Issue by gballet** [golangci/golangci-lint#924]
When the `misspell` linter tries to go over a line that is too long, it will show a warning. This warning will appear even if `misspell` is excluded for that file.

Thank you for creating the issue!

- [x] Yes, I'm using a binary release within 2 latest major releases. Only such installations are supported.
- [x] Yes, I've searched similar issues on GitHub and didn't find any.
- [x] Yes, I've included all information below (version, config, etc).

Please include the following information:

<details><summary>Version of golangci-lint</summary>

```console
$ golangci-lint --version
golangci-lint has version v1.22.2 built from (unknown, mod sum: "h1:iaihss3Tf6NvZVjun3lHimKSgofPV1+FqE/cbehoiRQ=") on (unknown)
```

</details>

<details><summary>Config file</summary>

```console
$ cat .golangci.yml
# This file configures github.com/golangci/golangci-lint.

run:
  timeout: 2m
  tests: true
  # default is true. Enables skipping of directories:
  #   vendor$, third_party$, testdata$, examples$, Godeps$, builtin$
  skip-dirs-use-default: true
  skip-files:
    - core/genesis_alloc.go

linters:
  disable-all: true
  enable:
    - deadcode
    - goconst
    - goimports
    - gosimple
    - govet
    - ineffassign
    - misspell
    # - staticcheck
    - unconvert
    # - unused
    - varcheck

linters-settings:
  gofmt:
    simplify: true
  goconst:
    min-len: 3 # minimum length of string constant
    min-occurrences: 6 # minimum number of occurrences

issues:
  exclude-rules:
    - path: crypto/blake2b/
      linters:
        - deadcode
    - path: crypto/bn256/cloudflare
      linters:
        - deadcode
    - path: p2p/discv5/
      linters:
        - deadcode
    - path: core/vm/instructions_test.go
      linters:
        - goconst
    - path: cmd/faucet/
      linters:
        - deadcode
    - path: core/genesis_alloc.go
      linters:
        - misspell
```

</details>

<details><summary>Go environment</summary>

```console
$ go version && go env
go version go1.13 linux/arm64
GO111MODULE=""
GOARCH="arm64"
GOBIN=""
GOCACHE="/home/gballet/.cache/go-build"
GOENV="/home/gballet/.config/go/env"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="arm64"
GOHOSTOS="linux"
GONOPROXY=""
GONOSUMDB=""
GOOS="linux"
GOPATH="/home/gballet/go"
GOPRIVATE=""
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/usr/local/go"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/usr/local/go/pkg/tool/linux_arm64"
GCCGO="gccgo"
AR="ar"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
GOMOD="/home/gballet/go/src/github.com/ethereum/go-ethereum/go.mod"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build350194322=/tmp/go-build -gno-record-gcc-switches"
```

</details>

<details><summary>Verbose output of running</summary>

```console
>>> build/cache/golangci-lint-1.22.2-linux-amd64/golangci-lint run --config .golangci.yml ./...

WARN [runner] Can't process result by autogenerated_exclude processor: can't filter issue result.Issue{FromLinter:"misspell", Text:"`ect` is a misspelling of `etc`", SourceLines:[]string(nil), Replacement:(*result.Replacement)(0xc00f0098f0), Pkg:(*packages.Package)(0xc001691770), LineRange:(*result.Range)(nil), Pos:token.Position{Filename:"core/genesis_alloc.go", Offset:0, Line:24, Column:64842}, HunkPos:0}: failed to get doc of file core/genesis_alloc.go: failed to scan file: bufio.Scanner: token too long 
```

</details>

**Comment by ernado** [comment]
Seems like was fixed when dealing with #954 in v1.23.3.
Please check again and feel free to reopen.
